src/components/filters/SurveyOptionFilter.jsx

```diff
@@ -30,7 +30,8 @@ export default class SurveyOptionFilter extends FilterBase {
     componentWillReceiveProps(nextProps) {
         this.setState(this.stateFromProps(nextProps));
 
-        if (nextProps.config.survey && nextProps.config.survey != this.props.config.survey) {
+        if (nextProps.config.survey && nextProps.config.survey != this.props.config.survey 
+                || !nextProps.elementsBySurvey[nextProps.config.survey]) {
             this.props.dispatch(retrieveSurvey(this.state.survey));
```

Comment:
The filter framework is really messy. I can't wait to move to Gen3. When I'm looking at this, I'm realizing that all of the conditions look at `nextProps` and `this.props`, but then when the action to retrieve the survey is dispatched, it uses  `this.state` to get the survey.

I guess that works if the `setState()` call on line 31 is synchronous, but I'm not sure it's guaranteed to be. Anyway, the change you've made makes sense!